src/cache_cf.cc

```diff
@@ -4873,11 +4939,43 @@ static void free_ftp_epsv(acl_access **ftp_epsv)
     FtpEspvDeprecated = false;
 }
 
+/// Like parseTimeLine() but does not require the timeunit to be specified.
+/// If missed, the default 'second' timeunit is assumed.
+static std::chrono::seconds
+ParseUrlRewriteTimeout()
+{
+    const auto valueToken = ConfigParser::NextToken();
+    if (!valueToken)
+        self_destruct();
+
+    using Seconds = std::chrono::seconds;
+
+    const auto parsedValue = xatof(valueToken);
+
+    std::chrono::nanoseconds parsedUnitDuration;
+
+    if (parsedValue) {
+        const auto unitToken = parsedValue ? ConfigParser::PeekAtToken() : nullptr;
+        if (parseTimeUnit<Seconds>(unitToken, parsedUnitDuration))
+            (void)ConfigParser::NextToken();
+        else {
+            const auto defaultParsed = parseTimeUnit<Seconds>(T_SECOND_STR, parsedUnitDuration);
+            assert(defaultParsed);
+            debugs(3, DBG_PARSE_NOTE(DBG_IMPORTANT), "WARNING: " << ConfigPosition << "missing time unit, using deprecated default '" << T_SECOND_STR << "'");
```

Comment:
This message does not make any sense. It is only reachable past the assert when parseTimeUnit was *successful* parsing the unit.

Reply:
Since Squid time parameters (including url_rewrite_timeout parameter) do not allow default time units, this should be a fatal error. However, I decided to leave the old behavior (the default 'second' time unit for url_rewrite_timeout) as is but add a deprecation warning, so that it would be easier to fix this in the future.